### PR TITLE
cdc: add ticdc integrity check golang implementation explanation.

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add a new section to introduce the Golang implementation

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: ticdc/ticdc-integrity-check.md [hunk 1]
```diff
@@ -90,15 +90,34 @@ fn checksum(columns) {
     * BIT、ENUM 和 SET 类型会被转换为 UINT64 类型。
 
         * BIT 类型按照二进制转换为 UINT64 类型。
-        * ENUM 和 SET 类型按照其对应的 INT 值转换为 UINT64 类型。例如，`SET('a','b','c')` 类型 column 的数据值为 `'a,c'`，则该值将被编码为 `0b101`。
+        * ENUM 和 SET 类型按照其对应的 INT 值转换为 UINT64 类型。例如，`SET('a','b','c')` 类型 column 的数据值为 `'a,c'`，则该值将被编码为 `0b101`，即 5。
```
Comment (Oreoxmt) [inline ticdc/ticdc-integrity-check.md:93]:
```suggestion
        * ENUM 和 SET 类型按照其对应的 INT 值转换为 UINT64 类型。例如，`SET('a','b','c')` 类型 column 的数据值为 `'a,c'`，则该值将被编码为 `0b101`，即 `5`。
```

File: ticdc/ticdc-integrity-check.md [hunk 2]
```diff
@@ -90,15 +90,34 @@ fn checksum(columns) {
...
 
     * TIMESTAMP、DATE、DURATION、DATETIME、JSON 和 DECIMAL 类型会被转换为 STRING 类型，然后转换为 UTF8 编码的字节。
-    * VARBIANRY、BINARY 和 BLOB（包括 TINY、MEDIUM 和 LONG）类型会直接使用它的字节。
-    * VARCHAR、CHAR 和 TEXT（包括 TINY、MEDIUM 和 LONG）类型会被编码为 UTF8 编码的字节。
+    * CHAR、VARCHAR、VARSTRING、STRING 和 Text、BLOB（包括 TINY、MEDIUM 和 LONG）等字符类型，会直接使用它的字节。
```
Comment (Oreoxmt) [inline ticdc/ticdc-integrity-check.md:96]:
```suggestion
    * CHAR、VARCHAR、VARSTRING、STRING 和 TEXT、BLOB（包括 TINY、MEDIUM 和 LONG）等字符类型，会直接使用它的字节。
```

File: ticdc/ticdc-integrity-check.md [hunk 3]
```diff
@@ -90,15 +90,34 @@ fn checksum(columns) {
...
+
+Checksum 计算和校验的过程如下：
+
+* 调用 [extractExpectedChecksum](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L281) 方法，拿到期望的 Checksum 值。如果该方法返回 false，则说明该事件不需要进行 Checksum 校验，因为上游并没有发送 Checksum。这可能发生在 TiCDC 开启了 Checksum，但是 TiDB 没有开启该功能，或者当前事件发生在 Checksum 校验功能开启之前等场景。
+* 调用 [calculateChecksum](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L461) 方法，遍历之前重建出来的所有列，使用每一列的 Value 和 MySQL Type，编码出一个字节切片，然后使用该字节切片，更新 Checksum 值。将具体数据编码为字节切片利用了 [buildChecksumBytes](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L482) 方法。
+* 通过 [verifyChecksum](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L444) 方法，进行 Checksum 计算和校验。将它和之前拿到的期望值进行比较，如果不相等，则说明 Checksum 校验失败，数据可能存在损坏的情况。
+
 > **注意：**
 >
 > 开启 Checksum 校验功能后，DECIMAL 和 UNSIGNED BIGINT 类型的数据会被转换为字符串类型。因此在下游消费者代码中需要将其转换为对应的数值类型，然后进行 Checksum 相关计算。
```
Comment (Oreoxmt) [inline ticdc/ticdc-integrity-check.md:122]:
```suggestion
> - 开启 Checksum 校验功能后，DECIMAL 和 UNSIGNED BIGINT 类型的数据会被转换为字符串类型。因此在下游消费者代码中需要将其转换为对应的数值类型，然后进行 Checksum 相关计算。
```

File: ticdc/ticdc-integrity-check.md [hunk 4]
```diff
@@ -90,15 +90,34 @@ fn checksum(columns) {
...
-
-Golang 消费者代码实现了解码从 Kafka 读取到的数据、按照 schema fields 排序以及 Checksum 计算等步骤。详情请参考 [`avro/decoder.go`](https://github.com/pingcap/tiflow/blob/master/pkg/sink/codec/avro/decoder.go)。
+> Delete 事件只含有 Handle Key 列的内容，而 Checksum 是基于所有列计算的，所以 Delete 事件不参与到 Checksum 的校验中。
```
Comment (Oreoxmt) [inline ticdc/ticdc-integrity-check.md:123]:
```suggestion
> - Delete 事件只含有 Handle Key 列的内容，而 Checksum 是基于所有列计算的，所以 Delete 事件不参与到 Checksum 的校验中。
```

File: ticdc/ticdc-integrity-check.md [hunk 5]
```diff
@@ -90,15 +90,34 @@ fn checksum(columns) {
...
+
+* 遍历完 `fields` 之后，就拿到了所有列数据的内容。对于 Delete 事件，将解码得到的 Columns 设置为 `PreColumns`，Insert 和 Update 事件都设置为 `Columns`。
+
+Checksum 计算和校验的过程如下：
+
+* 调用 [extractExpectedChecksum](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L281) 方法，拿到期望的 Checksum 值。如果该方法返回 false，则说明该事件不需要进行 Checksum 校验，因为上游并没有发送 Checksum。这可能发生在 TiCDC 开启了 Checksum，但是 TiDB 没有开启该功能，或者当前事件发生在 Checksum 校验功能开启之前等场景。
+* 调用 [calculateChecksum](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L461) 方法，遍历之前重建出来的所有列，使用每一列的 Value 和 MySQL Type，编码出一个字节切片，然后使用该字节切片，更新 Checksum 值。将具体数据编码为字节切片利用了 [buildChecksumBytes](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L482) 方法。
+* 通过 [verifyChecksum](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L444) 方法，进行 Checksum 计算和校验。将它和之前拿到的期望值进行比较，如果不相等，则说明 Checksum 校验失败，数据可能存在损坏的情况。
```
Comment (Oreoxmt) [inline ticdc/ticdc-integrity-check.md:118]:
```suggestion
1. 假设已经从 Kafka 读取到消息，并设置了 key 和 value 字段。分别对 key 和 value 进行解码操作，得到解码之后的数据值和 schema。具体的过程可以参考 [`decodeKey`](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L395) 和 [`decodeValue`](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L419) 方法。
2. 使用上一步解码得到的 key、value 和 schema 等内容，重新构建 `RowChangedEvent`。主要是构建每一列的数据内容，详情可见 [`assembleEvent`](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L176) 方法。构建 `RowChangedEvent` 的过程如下：

    1. 获取 schema 中所有的 `fields` 内容，遍历 `fields` 中的每一个元素 `field` 构建对应的列。其中 `fields` 已经按照 Column ID 排序。
    2. 利用 `field` 中包含的每一列的类型信息，重建每一列的 MySQL Type。并通过 keyMap 识别到 Handle Key 列，然后设置相应的 flag。
    3. valueMap 中的值需要经过 [`getColumnValue`](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L299) 转换，这是因为在编码过程中，某些列允许 NULL 存在，此时会把 value 编码成一个 map。因此在解码时需要从 map 中获取具体的值，即 map 中的第一个元素。如果该列是 `mysql.TypeEnum` 或 `mysql.TypeSet` 类型，还需要映射到它们的数字形式表示上。
    4. 遍历完 `fields` 后，就拿到了所有列数据的内容。对于 Delete 事件，将解码得到的 Columns 设置为 `PreColumns`，Insert 和 Update 事件都设置为 `Columns`。

Checksum 计算和校验的过程如下：

1. 调用 [`extractExpectedChecksum`](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L281) 方法，拿到期望的 Checksum 值。如果该方法返回 `false`，则说明该事件不需要进行 Checksum 校验，因为上游并没有发送 Checksum。这可能发生在 TiCDC 开启了 Checksum，但是 TiDB 没有开启该功能，或者当前事件发生在 Checksum 校验功能开启之前等场景。
2. 调用 [`calculateChecksum`](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L461) 方法，遍历之前重建出来的所有列。使用每一列的 value 和 MySQL Type，编码出一个字节切片。然后使用该字节切片更新 Checksum 值。将具体数据编码为字节切片利用了 [buildChecksumBytes](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L482) 方法。
3. 通过 [`verifyChecksum`](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L444) 方法，进行 Checksum 计算和校验。将步骤 2 计算的值与步骤 1 获取的期望值进行比较。如果不相等，则说明 Checksum 校验失败，数据可能存在损坏的情况。

```

如果这些流程是有顺序的，那么需要使用有序列表

File: ticdc/ticdc-integrity-check.md [hunk 6]
```diff
@@ -90,15 +90,32 @@ fn checksum(columns) {
...
+    4. 遍历完 `fields` 后，就拿到了所有列数据的内容。对于 Delete 事件，将解码得到的 Columns 设置为 `PreColumns`，Insert 和 Update 事件都设置为 `Columns`。
+
+Checksum 计算和校验的过程如下：
+
+1. 调用 [`extractExpectedChecksum`](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L281) 方法，拿到期望的 Checksum 值。如果该方法返回 `false`，则说明该事件不需要进行 Checksum 校验，因为上游并没有发送 Checksum。这可能发生在 TiCDC 开启了 Checksum，但是 TiDB 没有开启该功能，或者当前事件发生在 Checksum 校验功能开启之前等场景。
+2. 调用 [`calculateChecksum`](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L461) 方法，遍历之前重建出来的所有列。使用每一列的 value 和 MySQL Type，编码出一个字节切片。然后使用该字节切片更新 Checksum 值。将具体数据编码为字节切片利用了 [buildChecksumBytes](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L482) 方法。
+3. 通过 [`verifyChecksum`](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L444) 方法，进行 Checksum 计算和校验。将步骤 2 计算的值与步骤 1 获取的期望值进行比较。如果不相等，则说明 Checksum 校验失败，数据可能存在损坏的情况。
+
+
```
Comment (Oreoxmt) [inline ticdc/ticdc-integrity-check.md:117]:
```suggestion

```

File: ticdc/ticdc-integrity-check.md [hunk 7]
```diff
@@ -90,15 +90,32 @@ fn checksum(columns) {
...
+    * TIMESTAMP、DATE、DURATION、DATETIME、JSON 和 DECIMAL 类型会被转换为 STRING 类型，然后转换为字节。
+    * CHAR、VARCHAR、VARSTRING、STRING、TEXT、BLOB（包括 TINY、MEDIUM 和 LONG）等字符类型，会直接使用字节。
     * NULL 和 GEOMETRY 类型不会被纳入到 Checksum 计算中，返回空字节。
 
+## 基于 Golang 的 Avro 数据消费和 Checksum 计算过程解释
+
+TiCDC 提供了基于 Golang 的 Checksum 计算过程，你可以参考该过程实现自己的 Checksum 计算逻辑。主要代码逻辑位于 TiCDC Avro Decoder 实现的 [NextRowChangedEvent](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L100) 方法。该方法的具体工作过程如下：
+
+1. 假设已经从 Kafka 读取到消息，并设置了 key 和 value 字段。分别对 key 和 value 进行解码操作，得到解码之后的数据值和 schema。具体的过程可以参考 [`decodeKey`](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L395) 和 [`decodeValue`](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L419) 方法。
+2. 使用上一步解码得到的 key、value 和 schema 等内容，重新构建 `RowChangedEvent`。主要是构建每一列的数据内容，详情可见 [`assembleEvent`](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L176) 方法。构建 `RowChangedEvent` 的过程如下：
```
Comment (qiancai) [inline ticdc/ticdc-integrity-check.md:104]:
```suggestion
2. 使用解码后的 key、value 和 schema 等内容，重新构建每一列的数据内容 `RowChangedEvent`。详情可见 [`assembleEvent`](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L176) 方法。构建 `RowChangedEvent` 的过程如下：
```

File: ticdc/ticdc-integrity-check.md [hunk 8]
```diff
@@ -90,15 +90,32 @@ fn checksum(columns) {
...
+
+    1. 获取 schema 中所有的 `fields` 内容，遍历 `fields` 中的每一个元素 `field` 构建对应的列。其中 `fields` 已经按照 Column ID 排序。
```
Comment (qiancai) [inline ticdc/ticdc-integrity-check.md:106]:
```suggestion
    1. 获取 schema 中所有的 `fields` 内容，遍历 `fields` 中的每一个元素 `field` 以构建对应的列。其中 `fields` 已经按照 Column ID 排序。
```

File: ticdc/ticdc-integrity-check.md [hunk 9]
```diff
@@ -90,15 +90,32 @@ fn checksum(columns) {
...
+    2. 利用 `field` 中包含的每一列的类型信息，重建每一列的 MySQL Type。并通过 keyMap 识别到 Handle Key 列，然后设置相应的 flag。
+    3. valueMap 中的值需要经过 [`getColumnValue`](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L299) 转换，这是因为在编码过程中，某些列允许为 NULL，此时会把 value 编码成一个 map。因此在解码时需要从 map 中获取具体的值，即 map 中的第一个元素。如果该列是 `mysql.TypeEnum` 或 `mysql.TypeSet` 类型，还需要映射到它们的数字形式表示上。
```
Comment (qiancai) [inline ticdc/ticdc-integrity-check.md:108]:
```suggestion
    3. valueMap 中的值需要经过 [`getColumnValue`](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L299) 转换，这是因为在编码过程中，某些列允许为 NULL，此时会把 value 编码成一个 map。因此，解码时需要从 map 中获取具体的值，即 map 中的第一个元素。如果该列是 `mysql.TypeEnum` 或 `mysql.TypeSet` 类型，还需要映射到它们的数字形式表示上。
```

File: ticdc/ticdc-integrity-check.md [hunk 10]
```diff
@@ -90,15 +90,32 @@ fn checksum(columns) {
...
+    4. 遍历完 `fields` 后，就拿到了所有列数据的内容。对于 Delete 事件，将解码得到的 Columns 设置为 `PreColumns`，Insert 和 Update 事件都设置为 `Columns`。
```
Comment (qiancai) [inline ticdc/ticdc-integrity-check.md:109]:
```suggestion
    4. 遍历完 `fields` 后，就获取了所有列的数据。对于 Delete 事件，将解码得到的列设置为 `PreColumns`；对于 Insert 和 Update 事件，将解码得到的列设置为 `Columns`。
```

File: ticdc/ticdc-integrity-check.md [hunk 11]
```diff
@@ -90,15 +90,32 @@ fn checksum(columns) {
...
+
+Checksum 计算和校验的过程如下：
+
+1. 调用 [`extractExpectedChecksum`](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L281) 方法，拿到期望的 Checksum 值。如果该方法返回 `false`，则说明该事件不需要进行 Checksum 校验，因为上游并没有发送 Checksum。这可能发生在 TiCDC 开启了 Checksum，但是 TiDB 没有开启该功能，或者当前事件发生在 Checksum 校验功能开启之前等场景。
```
Comment (qiancai) [inline ticdc/ticdc-integrity-check.md:113]:
```suggestion
1. 调用 [`extractExpectedChecksum`](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L281) 方法，获取期望的 Checksum 值。如果该方法返回 `false`，则说明该事件不需要进行 Checksum 校验，因为上游并没有发送 Checksum。这可能发生在 TiCDC 开启了 Checksum 但 TiDB 没有开启该功能时，或者当前事件发生在 Checksum 校验功能开启之前等场景。
```

File: ticdc/ticdc-integrity-check.md [hunk 12]
```diff
@@ -90,15 +90,32 @@ fn checksum(columns) {
...
+2. 调用 [`calculateChecksum`](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L461) 方法，遍历之前重建出来的所有列。使用每一列的 value 和 MySQL Type，编码出一个字节切片。然后使用该字节切片更新 Checksum 值。将具体数据编码为字节切片利用了 [buildChecksumBytes](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L482) 方法。
```
Comment (qiancai) [inline ticdc/ticdc-integrity-check.md:114]:
```suggestion
2. 调用 [`calculateChecksum`](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L461) 方法，遍历之前重建出来的所有列。利用 [buildChecksumBytes](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L482) 方法将每一列的 value 和 MySQL Type 编码为一个字节切片，然后使用该字节切片更新 Checksum 值。
```

**Review by Oreoxmt — APPROVED** [review]
/lgtm

**Review by qiancai — APPROVED** [review]
/lgtm

File: ticdc/ticdc-integrity-check.md [hunk 13]
```diff
@@ -90,15 +90,30 @@ fn checksum(columns) {
...
+    4. 遍历完 `fields` 后，就获取了所有列的数据。对于 Delete 事件，将解码得到的列设置为 `PreColumns`；对于 Insert 和 Update 事件，将解码得到的列设置为 `Columns`。
+
+Checksum 计算和校验的过程如下：
+
+1. 调用 [`extractExpectedChecksum`](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L281) 方法，获取期望的 Checksum 值。如果该方法返回 `false`，则说明该事件不需要进行 Checksum 校验，因为上游并没有发送 Checksum。这可能发生在 TiCDC 开启了 Checksum 但 TiDB 没有开启该功能时，或者当前事件发生在 Checksum 校验功能开启之前等场景。
+2. 调用 [`calculateChecksum`](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L461) 方法，遍历之前重建出来的所有列。利用 [buildChecksumBytes](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L482) 方法将每一列的 value 和 MySQL Type 编码为一个字节切片，然后使用该字节切片更新 Checksum 值。
+3. 通过 [`verifyChecksum`](https://github.com/pingcap/tiflow/blob/eb04aecaf8e61f7f9d67597c2d2ef1f44583dd79/pkg/sink/codec/avro/decoder.go#L444) 方法，进行 Checksum 计算和校验。将步骤 2 计算的值与步骤 1 获取的期望值进行比较。如果不相等，则说明 Checksum 校验失败，数据可能存在损坏的情况。
 > **注意：**
```
Comment (qiancai) [inline ticdc/ticdc-integrity-check.md:116]:
```suggestion

> **注意：**
```

**Review by Oreoxmt — APPROVED** [review]

**Pull request merged**